ARCs/arc-0200.md

```diff
@@ -0,0 +1,232 @@
+      "readonly": false,
+      "args": [
+        {
+          "type": "address",
+          "name": "from_",
+          "desc": "The source  of the transfer"
+        },
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of the transfer"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "Amount of tokens to transfer"
+        }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "approve",
+      "desc": "Approve spender for a token",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "spender_" },
+        { "type": "uint256", "name": "value_" }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "allowance",
+      "desc": "Returns the current allowance of the spender of the tokens of the owner",
+      "readonly": true,
+      "args": [
+        { "type": "address", "name": "owner_" },
+        { "type": "address", "name": "spender_" }
+      ],
+      "returns": { "type": "uint256", "desc": "The remaining allowance" }
+    }
+  ],
+  "events": [
+    {
+      "name": "Transfer",
+      "desc": "Transfer of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "from_",
+          "desc": "The source of transfer of tokens"
+        },
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of transfer of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens transferred"
+        }
+      ]
+    },
+    {
+      "name": "Approval",
+      "desc": "Approval of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "owner_",
+          "desc": "The owner of the tokens"
+        },
+        {
+          "type": "address",
+          "name": "spender_",
+          "desc": "The approved spender of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens approve"
+        }
+      ]
+    }
+  ]
+}
+```
+
+Ownership of a token by a zero address indicates that a token is out of circulation indefinitely, or otherwise burned or destroyed.
+
+The methods `transfer` and `transferFrom` method MUST error when the balance of `from_` is insufficient.
+The `transferFrom` method MUST error unless called by an approved spender as defined by an extension defined in this ARC.
+The methods `transfer` and `transferFrom` MUST emit a `Transfer` event that a transfer is successful.
+A `Transfer` event SHOULD be emitted, with `from_` being the zero address, when a token is first minted.
+A `Transfer` event SHOULD be emitted, with `to_` being the zero address, when a token is destroyed.
+
+The `Approval` event MUST be emitted when the `approve` method is called successfully.
```

Comment:
Do you want to "fix" the Ethereum misfeature here, and put `increaseAllowance` and `decreaseAllowance` into the standard?
https://www.adrianhetman.com/unboxing-erc20-approve-issues/

Reply:
I may want to fix this issue from the get-go because the ARC-200 spec implementation may easily be ported to a EVM based chain with this fix already applied. I haven't heard of any issues with front running on Algorand yet but this seems like a great way to future-proof the spec. Will be considering.

Reply:
@jannotti I am on the side of not adding `increaseAllowance` and `decreaseAllowance` to the standard for simplicity sakes. I do think that implementations may add non-standard methods similar to the way OpenZeppelin does it with ERC-20 (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol)

I have a working ARC200 implementation that methods like  `increaseAllowance` and `decreaseAllowance` for convenience.